dlt/helpers/airflow_helper.py

```diff
@@ -291,8 +296,28 @@ def log_after_attempt(retry_state: RetryCallState) -> None:
                         pt >> nt
                     pt = nt
                 return tasks
+            elif decompose == "parallel":
+                if not isinstance(data, DltSource):
+                    raise ValueError("Can only decompose dlt sources")
+
+                if pipeline.full_refresh:
+                    raise ValueError("Cannot decompose pipelines with full_refresh set")
+
+                # parallel tasks
+                tasks = []
+                group_name = task_name(pipeline, data)
+                start = DummyOperator(task_id=f"{group_name}_start")
+
+                with TaskGroup(group_id=group_name):
+                    for source in data.decompose("scc"):
+                        tasks.append(make_task(pipeline, source))
+
+                end = DummyOperator(task_id=f"{group_name}_end")
+
+                start >> tasks >> end
+                return [start] + tasks + [end]
             else:
-                raise ValueError(decompose)
+                raise ValueError("decompose value must be one of ['none', 'serialize', 'parallel']")
```

Comment:
I've added a `TaskGroup()`, as it seems to me parallel tasks should be grouped - it'll make analyzes easier, both in the runtime and in the Airflow UI.

Reply:
do we have a task group within task group? mind `with self:` up there. I do not think we need another one. also it may make sense for you to run Airflow with the UI and simply see the dags you are creating in it.

Reply:
Well, yes, to indicate the group of those tasks executed in parallel to each other (as some tasks under `self` are sequential). Okay, dropped the group.